### PR TITLE
refactor: add support in Vite for loading `.react.svg` files as React components

### PR DESCRIPTION
**Pull request by emmenko**
Does the same thing [as the Webpack setup](https://github.com/commercetools/merchant-center-application-kit/blob/702a7b26b050c46215eed273a2d47b885deda611/packages/mc-scripts/src/config/create-webpack-config-for-development.ts#L238-L256).

### CODE REVIEW

File: packages/application-config/client.d.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
 type CSSModuleClasses = { readonly [key: string]: string };
 
+declare namespace NodeJS {
+  // Used for global variable replacements.
+  interface ProcessEnv {
+    readonly DEBUG: 'true' | 'false';
+    readonly NODE_ENV: 'development' | 'production' | 'test';
+  }
+}
```
Comment (emmenko) [inline packages/application-config/client.d.ts:9]: Slightly unrelated but noticed this in the CRA setup: https://github.com/facebook/create-react-app/blob/f34d88e30c7d8be7181f728d1abc4fd8d5cd07d3/packages/react-scripts/lib/react-app.d.ts#L5-L10

File: packages/application-config/client.d.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
 type CSSModuleClasses = { readonly [key: string]: string };
 
+declare namespace NodeJS {
+  // Used for global variable replacements.
+  interface ProcessEnv {
+    readonly DEBUG: 'true' | 'false';
+    readonly NODE_ENV: 'development' | 'production' | 'test';
```
Comment (tdeekens) [inline packages/application-config/client.d.ts:7]: Don't we assign a value to this somewhere? Maybe then it can't be readonly.

Comment (emmenko) [reply packages/application-config/client.d.ts:7]: Yes right, let me remove it then as it's then more confusing.

File: packages/mc-scripts/src/commands/start-vite.ts [hunk 3]
```diff
@@ -111,6 +112,26 @@ async function run() {
           plugins: ['@emotion/babel-plugin'],
         },
       }),
+      pluginSvgr({
```
Comment (CarlosCortizasCT) [inline packages/mc-scripts/src/commands/start-vite.ts:115]: I was wondering if we also need to include this plugin in the `build-vite` command as well.

Comment (emmenko) [reply packages/mc-scripts/src/commands/start-vite.ts:115]: Good point, yes we do. I forgot.

**Review by tdeekens — APPROVED** [review]

**Pull request merged**